### DISCUSSION TITLE
Soldering tweezers with modified STM32 Soldering Station Firmware

### DISCUSSION THREAD

**Discussion by jopl** [deividAlfa/stm32_soldering_iron_controller#128] [Ideas]
Dear STM32 Soldering Station Firmware enthusiastics,
if anyone is interested, except T12 compatible soldering station design ( https://jopl.eu/t12solderingstation.php ) I designed new one for soldering tweezers ( https://jopl.eu/solderingtweezers.php ). Thanks to David I was allowed to modify firmware, which is available at the site of design (just hex, as source is quite big and I am not so experienced to expose it on github as fork :-(, but I can share it, of course).
Regards, Josef.

**Comment by deividAlfa** [comment]
Very nice, but please don't make it closed source, you only need to fork the repo (Click fork, that's all), then upload the modified files replacing the original ones.
I don't think it'll take more than 5 minutes,

Thanks!

**Reply by deividAlfa** [reply]
Hello again. Please share the code for everyone!

**Comment by arekm** [comment]
"Thanks to David I was allowed to modify firmware" - did @jopl got special license for binary only release?

(thinking about GPL v3 license / copyright law violation)

**Comment by jopl** [comment]
Hi, no, just an allowance to modify code. But I am not able to publish (fork) code (I tried it several times), I am so sorry for that. I can provide the whole code, if you want ... send me some link to share ;-).
jopl

**Reply by deividAlfa** [reply]
Hi, you could simply drag & drop a .zip file here, no worries about the code quality, we're not NASA engineers 😄.
Just delete any Release / Debug folder, they take a lot of space.

**Comment by jopl** [comment]
[SolderingTweezers.zip](https://github.com/user-attachments/files/21455702/SolderingTweezers.zip)
Hi David, I am uploading zip file. It doesn't contain original folder "Drivers", as it has zipped approx. 41 MB and github has limit 25 MB per file :-(. Hopefully I hasn't change anything in this folder and the source will be helpful for everyone, who is interested ...
In case of any other need, don't hesitate to contact me, please.
Regards, Josef.

**Reply by deividAlfa** [reply]
How 41MB? Delete everything except Core folder, or the files you modified.

**Comment by jopl** [comment]
Of course, if you will be able to fork it on github somehow, I will be more than happy to see it live :-)

**Comment by arekm** [comment]
@jopl Or just place it on your web site (where you published hex file).

**Comment by jopl** [comment]
OK, I can. But am I allowed? David, please, confirm it. Thanks in advance.

**Reply by deividAlfa** [reply]
Of course you can! GPL says you can modify as you want, and share in any form.

**Comment by arekm** [comment]
GPL v3.0 allows you to modify code, publish modified code BUT your are obliged to provide source code, too (if you provide binaries like hex file). You don't have to even ask. That's how (in simple terms) that license works.

**Comment by jopl** [comment]
Ok, I'll publish it in the evening. Thx guys!

**Comment by jopl** [comment]
Hi friends,
the completed .zip file with all the source files is published on my site https://jopl.eu/solderingtweezers.php ...
https://jopl.eu/files/SolderingTweezers_Source.zip
Have a nice evening.
